httpd/filters.go

```diff
@@ -0,0 +1,147 @@
+package httpd
+
+import (
+	"encoding/base64"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+	"time"
+
+	"code.google.com/p/go-uuid/uuid"
+
+	"github.com/influxdb/influxdb"
+)
+
+// basicAuthCredentials returns the username and password encoded in
+// a request. The credentials may be present as URL query params, or as
+// a Basic Authentication header.
+func basicAuthCredentials(r *http.Request) (string, string, error) {
```

Comment:
Very true.  I was going to ask why we support passing username/password via parameters?  Basic auth is easy enough to do http://username:password@foo.com as a client.  Is this a backwards compatibility issue?  I'm not sure why we support taking parameters which duplicates basic auth functionality.